### PR TITLE
chore: add coverage badge to README

### PR DESCRIPTION
**Pull request by exoego**
Follow up to #2862 

### The author should do the following, if applicable

N/A, doc change only

- [ ] Add tests
- [ ] Run tests
- [ ] `bun run format:fix && bun run lint:fix` to format the code
- [ ] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 [![GitHub commit activity](https://img.shields.io/github/commit-activity/m/honojs/hono)](https://github.com/honojs/hono/pulse)
 [![GitHub last commit](https://img.shields.io/github/last-commit/honojs/hono)](https://github.com/honojs/hono/commits/main)
 [![Discord badge](https://img.shields.io/discord/1011308539819597844?label=Discord&logo=Discord)](https://discord.gg/KMh2eNSdxV)
+[![codecov](https://codecov.io/github/honojs/hono/graph/badge.svg)](https://codecov.io/github/honojs/hono)
```
Comment (yusukebe) [inline README.md:26]: It's nitpicking, but how about the following:

CleanShot 2024-06-01 at 11 14 39@2x

This is better categorised.

**Review by yusukebe — COMMENTED** [review]
@exoego 

I also want the badge! I've left a comment.

File: README.md (hunk 1 above)
Comment (exoego) [reply README.md:25]: Makes sense.
Done in https://github.com/honojs/hono/pull/2881/commits/6e4ee3e3966a46fa0e839236db0e18f6a9ce4f10

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@exoego 

Thanks!

**Pull request merged**